canonicalwebteam/blog/wordpress_api.py

```diff
@@ -19,71 +21,64 @@ def process_response(response):
     return response.json()
 
 
-def build_get_articles_url(
-    tags=[],
-    per_page=12,
-    page=1,
-    tags_exclude=[],
-    exclude=[],
-    categories=[],
-    sticky="",
-    groups=[],
-    after="",
-    before="",
-    author="",
-):
+def build_url(endpoint, params):
     """
     Build url to fetch articles from Wordpress api
-    :param tags: Array of tag ids to fetch articles for
-    :param per_page: Articles to get per page
-    :param page: Page number to get
-    :param tags_exclude: Array of IDs of tags that will be excluded
-    :param exclude: Array of article IDs to be excluded
-    :param category: Array of categories, which articles
-        should be fetched for
-    :param sticky: string 'true' or 'false' to only get featured articles
-    :param before: ISO8601 compliant date string to limit by date
-    :param after: ISO8601 compliant date string to limit by date
+    :param endpoint: The REST endpoint to fetch data from
+    :param params: Dictionary of parameter keys and their values
 
     :returns: URL to Wordpress api
     """
-    url = (
-        f"{API_URL}/posts?per_page={per_page}"
-        f"&tags={','.join(str(id) for id in tags)}"
-        f"&page={page}"
-        f"&group={','.join(str(id) for id in groups)}"
-        f"&tags_exclude={','.join(str(id) for id in tags_exclude)}"
-        f"&categories={','.join(str(id) for id in categories)}"
-        f"&exclude={','.join(str(id) for id in exclude)}"
-        f"&author={author}&_embed"
-    )
-    if sticky != "":
-        url = url + f"&sticky={sticky}"
-    if before != "":
-        url = url + f"&before={before}"
-    if after != "":
-        url = url + f"&after={after}"
+    params_object = [param for param in params.items() if param[1]]
```

Comment:
I don't think this does what you want:

``` python3
In [2]: params = {'tags': [1, 2, 3, 4]}

In [3]: params_object = [param for param in params.items() if param[1]]

In [4]: params_object
Out[4]: [('tags', [1, 2, 3, 4])]

In [5]: from urllib.parse import urlencode

In [6]: from urllib.parse import urlencode, unquote

In [7]: unquote(urlencode(params_object))
Out[7]: 'tags=[1,+2,+3,+4]'
```

I think what you want is more along the lines of:

``` python3
for key, value in params:
    if type(value) is list:
        params[key] = ",".join([str(item) for item in value])

query = urlencode(params_object)
```

Reply:
Oh yeah! Saves a lot of repition. 
Ive adjusted it slightly to only run on parameters that actually have values.